src/main/java/uk/gov/gchq/palisade/util/ResourceBuilder.java

```diff
@@ -134,6 +133,7 @@ public static Resource create(final String uriString, final Map<String, String>
         try {
             return create(new URI(uriString), attributes);
         } catch (URISyntaxException ex) {
+            LOGGER.error("Resource create threw an error when creating a URI ", ex);
             throw new IllegalArgumentException("URISyntaxException converting string '" + uriString + "' to uri");

```

Comment:
Instead of logging the error and throwing an exception, can you just add the exception to the illegal argument exception
```suggestion
            throw new IllegalArgumentException("URISyntaxException converting string '" + uriString + "' to uri", ex);
```

Reply:
done